File: problems/129/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumNumbers(TreeNode* root) {
        if (not root) {
            return 0;
        }
        
        int ans = 0;
        queue<pair<TreeNode*, int>> q;
        
        q.push({root, 0});
        
        while (not empty(q)) {
            auto [node, curVal] = q.front();
            q.pop();
            
            curVal = curVal * 10 + node->val;
            
            if (not node->left and not node->right) {
                ans += curVal;
            }
            
            if (node->left) {
                q.push({node->left, curVal});
            }
            
            if (node->right) {
                q.push({node->right, curVal});
            }
        }
        
        return ans;
    }
};

File: problems/129/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: TreeNode) -> int:
        if not root:
            return 0
        
        q, s = deque([(root, 0)]), 0
        
        while q:
            for _ in range(len(q)):
                node, cs = q.popleft()
                curs = cs * 10 + node.val
                
                if node.left:
                    q.append((node.left, curs))
                
                if node.right:
                    q.append((node.right, curs))
                
                if not node.left and not node.right:
                    s += curs
        
        return s